### PR TITLE
record: handle files refactoring

### PR DESCRIPTION
**Pull request by drjova**
Signed-off-by: Harris Tzovanakis <me@drjova.com>

### CODE REVIEW

File: backend/inspirehep/records/api/literature.py [hunk 1]
```diff
@@ -160,93 +164,40 @@ def update(
                 push_to_orcid(self)
             self.push_authors_phonetic_blocks_to_redis()
 
-    def set_files(self, documents=None, figures=None, force=False):
+    def add_files(self):
```
Comment (pazembrz) [inline backend/inspirehep/records/api/literature.py]: Add some doc-strings with explanation how this works now.

File: backend/inspirehep/records/api/mixins.py [hunk 2]
```diff
@@ -306,160 +277,91 @@ def _download_file_from_url(self, url):
...
-            )
-            .one_or_none()
-        )
-        if bucket:
-            LOGGER.debug("Bucket found", key=bucket.bucket.id, uuid=self.id)
-            return bucket.bucket
-        LOGGER.info("Bucket not found", uuid=self.id)
-        return self._create_bucket(location, storage_class)
-
-    def _create_bucket(self, location=None, storage_class=None):
```
Comment (pazembrz) [inline backend/inspirehep/records/api/mixins.py]: They are still saying to implement your own _create_bucket
https://github.com/inveniosoftware/invenio-records-files/blob/ad9bd1dfbd02eb745e17d2cc23a1dc3583e6dbcc/invenio_records_files/api.py#L242-L243

And actually their create_bucket method is just silly...

Comment (drjova) [reply backend/inspirehep/records/api/mixins.py]: This is the subclass :) https://github.com/inveniosoftware/invenio-records-files/blob/ad9bd1dfbd02eb745e17d2cc23a1dc3583e6dbcc/invenio_records_files/api.py#L325

File: backend/inspirehep/records/api/literature.py [hunk 3]
```diff
@@ -160,93 +164,40 @@ def update(
...
-        return files
+        for figure in figures:
+            metadata = self.add_file(**figure)
+            builder.add_figure(**metadata)
+            keys.append(metadata["key"])
+
+            if "figures" in builder.record:
+                self["figures"] = builder.record["figures"]
+
+        for key in list(self.files.keys):
```
Comment (pazembrz) [inline backend/inspirehep/records/api/literature.py]: And what if self.files are None?

**Comment by drjova** [comment]
@pazembrz updated

**Pull request merged**